package/tests/test_server.py

```diff
@@ -188,3 +189,52 @@ def test_nodes_endpoint(client):
     assert response.status_code == 200
     data = json.loads(response.data.decode())
     assert data == EXPECTED_PIPELINE_DATA
+
+
+@pytest.fixture(autouse=True)
+def clean_up():
+    server.VIZ_THREADS = {}
```

Comment:
It might be better to do this in case something holds the reference. 
```suggestion
    server.VIZ_THREADS.clear()
```